README.md

```diff
@@ -63,6 +63,20 @@ export type MyNode = { firstName: string; lastName: string; age: number };
 
 export type MyEdge = { items: number[] };
 
+// Define codec for encoding and decoding Id to string
+
+export const MyIdCodec: Codec<string, string, MyId> = {
+  encode: (a: MyId) => a.toString(),
+  decode: (i: string) =>
+    pipe(
+      Number(i),
```

Comment:
It would be nice if we could somehow not confront new users with such parsing details.
I don't know if it's much better, but the following should work (not tested though)
```ts
export const MyIdCodec: Codec<string, string, MyId> = compose(string)(number)
```

`compose(string)(number)` is of type `Codec<unknown, number, string>` and it's always valid to assert a stricter type, so unknown can be just replaced with string.

Reply:
I'll make empty string fail, however you can't use `compose` in this way:
```
const compose: <L, A extends L, P extends A, B>(to: Codec<L, P, B>) => <I, O>(from: Codec<I, O, A>) => Codec<I, O, B>
number: Codec<unknown, number, number> => I = unknown, O = number, A = number
string: Codec<unknown, string, string> => L = unknown, P = string, B = string

P = string extends A = number
```

Reply:
ok, I see. I did not care about the "extends" part. ok, that does not work. but my main point was to make it somehow simple. Not everyone is familiar with codecs.

I think I'd prefer to use string as Id in the examples, then we can use a simple identity codec. And we can add another ("advanced") section, that explains how to use different types as Id.

For many people it will be just fine to use string.

Reply:
See the updated examples where I use plain `string` as `Id` type.